component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java

```diff
@@ -1,62 +1,90 @@
-      CalendarEventSearchQuery.class);
+  public CalendarEventSearchQuery(WikiReference wikiRef) {
+    super(wikiRef);
+  }
+
+  public CalendarEventSearchQuery(IEventSearchQuery query) {
+    super(query, null);
+  }
 
-  private final Date startDate;
-  private final boolean isArchive;
-  private final List<String> allowedSpaces;
-  
+  @Deprecated
   public CalendarEventSearchQuery(ICalendar cal, List<String> sortFields) {
     super(cal.getWikiRef(), getDefaultSortFields(sortFields, cal.isArchive()));
     this.startDate = cal.getStartDate();
     this.isArchive = cal.isArchive();
     this.allowedSpaces = cal.getAllowedSpaces();
   }
 
+  @Deprecated
   public CalendarEventSearchQuery(ICalendar cal, IEventSearchQuery query) {
     super(query, getDefaultSortFields(null, cal.isArchive()));
     this.startDate = cal.getStartDate();
     this.isArchive = cal.isArchive();
     this.allowedSpaces = cal.getAllowedSpaces();
   }
 
+  @Override
+  public void addCalendarRestrictions(ICalendar cal) {
+    this.startDate = cal.getStartDate();
+    this.isArchive = cal.isArchive();
+    this.allowedSpaces = cal.getAllowedSpaces();
+    setSortFields(getDefaultSortFields(getSortFields(), cal.isArchive()));
+  }
+
   @Override
   protected LuceneQuery getAsLuceneQueryInternal(LuceneQuery query) {
     query = super.getAsLuceneQueryInternal(query);
-    query.add(getSearchService().createRestrictionGroup(Type.OR, Arrays.asList("space"), 
+    return addCalendarRestrictions(query);
+  }
+
+  private LuceneQuery addCalendarRestrictions(LuceneQuery query) {
+    checkNotNull("Calendar must be initialized before calling addCalendarRestrictions",
+        this.isArchive);
+    query.add(getSearchService().createRestrictionGroup(Type.OR, Arrays.asList("space"),
         getAllowedSpaces()));
     if (!isArchive) {
       query.add(getSearchService().createFromDateRestriction(
-          ICalendarClassConfig.CALENDAR_EVENT_CLASS + "." 
-          + ICalendarClassConfig.PROPERTY_EVENT_DATE, startDate, true));
+          ICalendarClassConfig.CALENDAR_EVENT_CLASS + "."
+              + ICalendarClassConfig.PROPERTY_EVENT_DATE, startDate, true));
     } else {
-      query.add(getSearchService().createToDateRestriction(
-          ICalendarClassConfig.CALENDAR_EVENT_CLASS + "." 
-          + ICalendarClassConfig.PROPERTY_EVENT_DATE, startDate, false));
+      query.add(getSearchService().createToDateRestriction(ICalendarClassConfig.CALENDAR_EVENT_CLASS
+          + "." + ICalendarClassConfig.PROPERTY_EVENT_DATE, startDate, false));
     }
     return query;
   }
 
   private List<String> getAllowedSpaces() {
-    List<String> ret = new ArrayList<String>();
-    if (allowedSpaces != null && allowedSpaces.size() > 0) {
-      for(String space : allowedSpaces) {
+    List<String> ret = new ArrayList<>();
+    if ((allowedSpaces != null) && (allowedSpaces.size() > 0)) {
+      for (String space : allowedSpaces) {
         ret.add("\"" + space + "\"");
       }
```

Comment:
this can be replaced with `com.celements.search.lucene.LuceneUtils.exactify(List<String>)`.